readthedocs/proxito/middleware.py

```diff
@@ -176,6 +176,51 @@ def add_proxito_headers(self, request, response):
         else:
             response['X-RTD-Version-Method'] = 'path'
 
+    def add_user_headers(self, request, response):
+        """
+        Set specific HTTP headers requested by the user.
+
+        The headers added come from ``projects.models.HTTPHeader`` associated
+        with the ``Domain`` object.
+        """
+        if hasattr(request, 'domain'):
+            domain = request.domain
+            for http_header in domain.http_headers.all():
+                response[http_header.name] = http_header.value
```

Comment:
Idea: Should we warn if the header is already set? Not sure how often this would happen, but based on David's feedback about not double-setting them. I think this will override an existing value, but we probably want to check to see if we're clobbering anything.

Reply:
Cool! I'm `log.exception` here so we are aware of this.